code/modules/projectiles/updated_projectiles/guns/energy.dm

```diff
@@ -87,34 +113,12 @@
 	zoomdevicename = "scope"
 	flags_equip_slot = SLOT_BACK
 	w_class = 5
-	var/charge_time = 0
-	var/last_regen = 0
+	charge_cost = 100
 	flags_gun_features = GUN_UNUSUAL_DESIGN
 
-
 /obj/item/weapon/gun/energy/plasmarifle/New()
-	..()
-	processing_objects.Add(src)
-	last_regen = world.time
-	update_icon()
-	verbs -= /obj/item/weapon/gun/verb/field_strip
-	verbs -= /obj/item/weapon/gun/verb/toggle_burst
-	verbs -= /obj/item/weapon/gun/verb/empty_mag
-
-
-
-/obj/item/weapon/gun/energy/plasmarifle/Dispose()
 	. = ..()
-	processing_objects.Remove(src)
-
-
-/obj/item/weapon/gun/energy/plasmarifle/process()
```

Comment:
why did you remove the self-charging?  undocumented change

Reply:
It was a snowflake mechanic made for preds, not OOP or modular. All energy weapons now use cells properly.